Build failure: nufraw

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Build `nufraw` package.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/316629846

### Relevant log output

```console
nufraw_ui.c: In function 'input_fill_interface':
nufraw_ui.c:3414:94: error: passing argument 3 of 'curveeditor_widget_new' from incompatible pointer type [-Wincompatible-pointer-types]
 3414 |         gUiData.BaseCurveWidget = curveeditor_widget_new(UI_CURVE_HEIGHT, gUiData.LeftSpace, curve_update, (gpointer)base_curve);
      |                                                                                              ^~~~~~~~~~~~
      |                                                                                              |
      |                                                                                              void (*)(GtkWidget *, long int) {aka void (*)(struct _GtkWidget *, long int)}
In file included from nufraw_ui.c:20:
curveeditor_widget.h:15:72: note: expected 'void (*)(void)' but argument is of type 'void (*)(GtkWidget *, long int)' {aka 'void (*)(struct _GtkWidget *, long int)'}
   15 | extern GtkWidget *curveeditor_widget_new(int height, int width, void (*callback)(), gpointer userdata);
      |                                                                 ~~~~~~~^~~~~~~~~~~
nufraw_ui.c:1872:13: note: 'curve_update' declared here
 1872 | static void curve_update(GtkWidget *widget, long curveType)
      |             ^~~~~~~~~~~~
nufraw_ui.c: In function 'corrections_fill_interface':
nufraw_ui.c:3640:90: error: passing argument 3 of 'curveeditor_widget_new' from incompatible pointer type [-Wincompatible-pointer-types]
 3640 |         gUiData.CurveWidget = curveeditor_widget_new(UI_CURVE_HEIGHT, gUiData.LeftSpace, curve_update, (gpointer)luminosity_curve);
      |                                                                                          ^~~~~~~~~~~~
      |                                                                                          |
      |                                                                                          void (*)(GtkWidget *, long int) {aka void (*)(struct _GtkWidget *, long int)}
curveeditor_widget.h:15:72: note: expected 'void (*)(void)' but argument is of type 'void (*)(GtkWidget *, long int)' {aka 'void (*)(struct _GtkWidget *, long int)'}
   15 | extern GtkWidget *curveeditor_widget_new(int height, int width, void (*callback)(), gpointer userdata);
      |                                                                 ~~~~~~~^~~~~~~~~~~
nufraw_ui.c:1872:13: note: 'curve_update' declared here
 1872 | static void curve_update(GtkWidget *widget, long curveType)
      |             ^~~~~~~~~~~~
make[2]: *** [Makefile:883: nufraw_ui.o] Error 1
make[2]: Leaving directory '/build/nufraw-0.43-3'
make[1]: *** [Makefile:1075: all-recursive] Error 1
make[1]: Leaving directory '/build/nufraw-0.43-3'
make: *** [Makefile:624: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/xzl5dhddgjxp2p18h5qpkhvlkp8128j1-source`

### Notify maintainers

@asbachb 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/graphics/nufraw/default.nix
+++ b/pkgs/applications/graphics/nufraw/default.nix
@@ -56,7 +56,7 @@ stdenv.mkDerivation rec {
     "--enable-dst-correction"
   ];
 
-  env.NIX_CFLAGS_COMPILE = "-Wno-narrowing";
+  env.NIX_CFLAGS_COMPILE = "-Wno-narrowing -Wno-error=incompatible-pointer-types -std=gnu17";
 
   postInstall = lib.optionalString addThumbnailer ''
     mkdir -p $out/share/thumbnailers

